tests/test_cli.py

```diff
@@ -3,54 +3,250 @@
+        assert 'origin: ' + device_id in str(output)
+        assert '{"machine":{"temperature":' in str(output)
+    finally:
+        cli_stop(runner)
+        reset_docker_environment()
+
+
+def test_cli_setup_during_first_time(runner):
+    iniFile = HostPlatform.get_setting_ini_path()
+    jsonFile = HostPlatform.get_config_file_path()
+    update_setting_ini_as_firsttime()
+    if os.path.exists(iniFile):
+        os.remove(iniFile)
+    if os.path.exists(jsonFile):
+        os.remove(jsonFile)
+    cli_setup(runner)
+    assert os.path.exists(iniFile)
+    assert os.path.exists(jsonFile)
 
 
 def test_cli_modulecred(runner):
+    iothub_name = get_iothub_name_from_device_connection_string()
+    cli_setup(runner)
     result = runner.invoke(cli.modulecred)
     assert not result.exception
     assert result.exit_code == 0
-    assert 'EdgeHubConnectionString' in result.output.strip()
+    assert 'EdgeHubConnectionString=HostName=' + iothub_name in result.output.strip()
+    assert 'EdgeModuleCACertificateFile' in result.output.strip()
+    assert 'edge-device-ca.cert.pem' in result.output.strip()
 
 
-def test_cli_start(runner):
-    result = runner.invoke(cli.start, ['-i', 'input1'])
-    assert not result.exception
-    assert result.exit_code == 0
-    assert 'IoT Edge Simulator has been started in single module mode' in result.output.strip()
+def test_cli_output_modulecred_file(runner):
+    try:
+        iothub_name = get_iothub_name_from_device_connection_string()
+        cli_setup(runner)
+        modulered_output_file = 'modulecred_output.txt'
+        result = runner.invoke(cli.modulecred, ['-o', modulered_output_file])
+        output_file_path = os.path.join(workingdirectory, modulered_output_file)
+        assert not result.exception
+        assert result.exit_code == 0
+        assert os.path.exists(output_file_path)
+        with open(output_file_path) as f:
+            content = f.readlines()
+            assert 'EdgeHubConnectionString=HostName=' + iothub_name in str(content)
+            assert 'EdgeModuleCACertificateFile' in str(content)
+            assert 'edge-device-ca.cert.pem' in result.output.strip()
+    finally:
+        if os.path.exists(output_file_path):
+            os.remove(output_file_path)
 
 
-def test_cli_stop(runner):
-    result = runner.invoke(cli.stop)
-    assert not result.exception
-    assert result.exit_code == 0
-    assert 'IoT Edge Simulator has been stopped successfully' in result.output.strip()
+@pytest.mark.skipif(docker_client.get_os_type().lower() == 'windows', reason='It does not support windows container')
+def test_cli_create_options_for_custom_volume(runner):
+    try:
+        cli_setup(runner)
+        remove_docker_volume('testVolume')
+        cli_start_with_deployment(runner, 'deployment_with_custom_volume.json')
+        wait_verify_docker_output(['docker', 'volume', 'ls'], ['testVolume'])
+        expected_volumes = (['"Source": "testVolume"',
```

Comment:
we should also have edgehubdev volume

Reply:
> edgeHubDev container

add step to verify edgeHubDev container via docker inspect edgehubdev